R/pkg/R/sparkR.R

```diff
@@ -606,7 +606,7 @@ getClientModeSparkSubmitOpts <- function(submitOps, sparkEnvirMap) {
     # process only if --option is not already specified
     if (!is.null(opsValue) &&
         nchar(opsValue) > 1 &&
-        !grepl(sparkConfToSubmitOps[[conf]], submitOps)) {
+        !grepl(sparkConfToSubmitOps[[conf]], submitOps, fixed = TRUE)) {
```

Comment:
This could actually be a bug fix too for very, very unlikely corner cases such as `spark.driver.memory` <> `sparkAdriverXmemory`

Reply:
yes exactly. there were a few other like this in `utils.R` e.g.

`grep("org.apache.spark.sql.streaming.StreamingQueryException: "`